### PR TITLE
#1260 convert fileUtils to kotlin and prevent crash by checking the c…

### LINKED ISSUE #1260: Crash Report: IllegalArgumentException FileUtils.contentQuery (FileUtils.java:159)
**Describe the bug**
```
java.lang.RuntimeException: 

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2756)

 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2825)

 
  at android.app.ActivityThread.-wrap12 (ActivityThread.java)

 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1557)

 
  at android.os.Handler.dispatchMessage (Handler.java:110)

 
  at android.os.Looper.loop (Looper.java:203)

 
  at android.app.ActivityThread.main (ActivityThread.java:6339)

 
  at java.lang.reflect.Method.invoke (Native Method)

 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1084)

 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:945)
Caused by: java.lang.IllegalArgumentException: 

 
  at android.database.AbstractCursor.getColumnIndexOrThrow (AbstractCursor.java:333)

 
  at android.database.CursorWrapper.getColumnIndexOrThrow (CursorWrapper.java:87)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:159)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:144)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.manageExternalLaunchAndRestoringViewState (KiwixMobileActivity.java:1771)

 
  at org.kiwix.kiwixmobile.KiwixMobileActivity.onCreate (KiwixMobileActivity.java:462)

 
  at android.app.Activity.performCreate (Activity.java:6689)

 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1140)

 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2709)
```

### PR DESCRIPTION
**Pull request by macgills**
…olumn name exists before accessing it

Fixes #1260 

Changes: Kotlin conversion and Cursor column checking

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -230,7 +230,8 @@ android {
     checkAllWarnings true
     warningsAsErrors true
     ignore 'MissingTranslation', //TODO stop ignoring
-        'CheckResult'
+        'CheckResult',
+        'LogConditional'
```
Comment (ISNIT0) [inline app/build.gradle]: Why are we ignoring `LogConditional` here?

Comment (macgills) [reply app/build.gradle]: Well at no point thus far have we cared about logging in production.
The TODO here should really have an associated ticket where we go through and actually improve these.
Our own dedicated logging class would be nice to have.

Comment (ISNIT0) [reply app/build.gradle]: I'm tagging @julianharty because I have a feeling he'll find this amusing 😉

Comment (ISNIT0) [reply app/build.gradle]: We have a logging ticket here: https://github.com/kiwix/kiwix-android/issues/252

Can we get a ticket for this TODO? then I'd be happy with the change 👍

Comment (macgills) [reply app/build.gradle]: https://github.com/kiwix/kiwix-android/issues/1270
This is for specifically the conditional logging, I'll make a Lint tech debt ticket

Comment (macgills) [reply app/build.gradle]: https://github.com/kiwix/kiwix-android/issues/1271 adding the issues to it now

**Comment by kelson42** [comment]
Unfortunately, new code seems not unit tested at all (code coverage goes seriously back)

**Comment by macgills** [comment]
@kelson42 still a result of instrumentation. if you look at the change I have made it affected 5 files, codecov reports 9 files

**Pull request merged**